components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -1127,6 +1107,8 @@ SUBROUTINE p3_main(qc,nc,qr,nr,th,qv,dt,qitot,qirim,nitot,birim,   &
     !--These will be added as namelist parameters in the future
     logical(btype), parameter :: debug_ON     = .true.  !.true. to switch on debugging checks/traps throughout code  TODO: Turn this back off as default once the tlay error is found.
     logical(btype), parameter :: debug_ABORT  = .false.  !.true. will result in forced abort in s/r 'check_values'
+    
+    real(rtype),dimension(its:ite,kts:kte) :: qc_old, nc_old, qr_old, nr_old, qitot_old, nitot_old, qv_old, th_old
```

Comment:
As mentioned above, I think these *_old variables can be local to p3_main_main_loop?

Reply:
No, the qc,nc, etc ... are changing to in-cloud values in `p3_main_pre_main_loop`

Reply:
OK